agent/app/service/container.go

```diff
@@ -1475,3 +1472,39 @@ func loadComposeCount(client *client.Client) int {
 
 	return len(composeMap)
 }
+func loadContainerPortForInfo(itemPorts []types.Port) []dto.PortHelper {
+	var exposedPorts []dto.PortHelper
+	samePortMap := make(map[string]dto.PortHelper)
+	ports := transPortToStr(itemPorts)
+	var itemPort dto.PortHelper
+	for _, item := range ports {
+		itemStr := strings.Split(item, "->")
+		if len(itemStr) < 2 {
+			continue
+		}
+		lastIndex := strings.LastIndex(itemStr[0], ":")
+		if lastIndex == -1 {
+			itemPort.HostPort = itemStr[0]
+		} else {
+			itemPort.HostIP = itemStr[0][0:lastIndex]
+			itemPort.HostPort = itemStr[0][lastIndex+1:]
+		}
+		itemContainer := strings.Split(itemStr[1], "/")
+		if len(itemContainer) != 2 {
+			continue
+		}
+		itemPort.ContainerPort = itemContainer[0]
+		itemPort.Protocol = itemContainer[1]
+		keyItem := fmt.Sprintf("%s->%s/%s", itemPort.HostPort, itemPort.ContainerPort, itemPort.Protocol)
+		if val, ok := samePortMap[keyItem]; ok {
+			val.HostIP = ""
+			samePortMap[keyItem] = val
+		} else {
+			samePortMap[keyItem] = itemPort
+		}
+	}
+	for _, val := range samePortMap {
+		exposedPorts = append(exposedPorts, val)
+	}
+	return exposedPorts
+}
```

Comment:
There seems to be no need for further checks or optimizations at this point as the provided code appears fine. If there's anything specific you want checked regarding functionality or performance, please let me know!